tket/src/Mapping/LexiRoute.cpp

```diff
@@ -521,7 +521,50 @@ bool LexiRoute::solve(unsigned lookahead) {
   if (!check.first && !check.second) {
     // update circuit with new swap
     // final_labelling is initial labelling permuted by single swap
-    this->mapping_frontier_->add_swap(chosen_swap.first, chosen_swap.second);
+
+    // if false, SWAP is identical to last SWAP added without any gates realised
+    if (!this->mapping_frontier_->add_swap(
```

Comment:
Are there tests that exercise this? (I guess not since it was getting stuck before?) Would be good to add one.

Reply:
The scenario is extremely hard to replicate "naturally": it happened once in the thousands of circuits benchmarked very deep into a very large circuit. I'm fairly happy with the logic that if this is false we have a known workaround and that we know it calls false under the right circumstance (what 526 covers).

Reply:
As I'm still running benchmarks I'll generate and benchmark a number of small circuits with what I consider to be the potential to clock this case (i.e. just very dense circuits) and if one hits it that is somewhat small I'll a) double check it would have failed previously and then b) add it as a test case here with the criteria being that it successfully finishes routing.

Reply:
Sounds good; you can tag the test as "long" if it takes a long time (> 1s) to run.

Reply:
"not done" <- still churning through circuits

Reply:
OK. Am keen to have at least one verified run of the new code before merging. Thanks!